torch/onnx/_internal/exporter/_compat.py

```diff
@@ -105,8 +105,10 @@ def export_compat(
     dynamic_shapes_with_export_dim, need_axis_mapping = (
         _dynamic_shapes.convert_str_to_export_dim(dynamic_shapes)
     )
-
-    registry = _registration.ONNXRegistry.from_torchlib()
+    if opset_version is not None:
+        registry = _registration.ONNXRegistry().from_torchlib(opset_version=opset_version)
+    else:
+        registry = _registration.ONNXRegistry().from_torchlib()
```

Comment:
If opset_version is set during torchlib registration, do we still need model-wise opset conversion? https://github.com/pytorch/pytorch/blob/efc975feb26ee31649cfae2aaba62c1807ee98a6/torch/onnx/_internal/exporter/_compat.py#L180

Reply:
I think so- not every implementation is for the specified opset. Aka we can create a graph with mixed opsets then reconcile them with the version converter

Reply:
Good question ... I think we need it. If a new opset introduces an op-change that can be easily handled by version-converter (a common case), then no update is needed in torchlib. Updates to torchlib (using this PR feature) is required only when a new opset introduces new functionality that we couldn't use earlier (eg., new 4-bit types or new attention ops).